src/jit/emitxarch.cpp

```diff
@@ -4888,19 +4916,54 @@ void emitter::emitIns_AX_R(instruction ins, emitAttr attr, regNumber ireg, regNu
 }
 
 #if FEATURE_HW_INTRINSICS
-void emitter::emitIns_SIMD_R_R_R(instruction ins, regNumber reg, regNumber reg1, regNumber reg2, var_types simdtype)
+void emitter::emitIns_SIMD_R_R_R(instruction ins, regNumber reg, regNumber reg1, regNumber reg2, emitAttr attr)
+{
+    if (UseVEXEncoding())
+    {
+        emitIns_R_R_R(ins, attr, reg, reg1, reg2);
+    }
+    else
+    {
+        if (reg1 != reg)
+        {
+            emitIns_R_R(INS_movaps, attr, reg, reg1);
+        }
+        emitIns_R_R(ins, attr, reg, reg2);
+    }
+}
+
+void emitter::emitIns_SIMD_R_R_R_R(
+    instruction ins, regNumber reg, regNumber reg1, regNumber reg2, regNumber reg3, emitAttr attr)
 {
-    if (UseVEXEncoding() && reg1 != reg)
+    assert(ins == INS_vblendvps || ins == INS_vblendvpd || ins == INS_vpblendvb || ins == INS_blendvps ||
+           ins == INS_blendvpd || ins == INS_pblendvb);
+    if (UseVEXEncoding())
     {
-        emitIns_R_R_R(ins, emitTypeSize(simdtype), reg, reg1, reg2);
+        // convert SSE encoding of SSE4.1 instructions to VEX encoding
+        switch (ins)
+        {
+            case INS_blendvps:
+                ins = INS_vblendvps;
+            case INS_blendvpd:
+                ins = INS_vblendvpd;
+            case INS_pblendvb:
+                ins = INS_vpblendvb;
+        }
+        emitIns_R_R_R_R(ins, attr, reg, reg1, reg2, reg3);
     }
     else
     {
+        assert(ins == INS_blendvps || ins == INS_blendvpd || ins == INS_pblendvb);
+        // SSE4.1 blendv* hardcode the mask vector in XMM0
+        if (reg3 != REG_XMM0)
+        {
+            emitIns_R_R(INS_movaps, attr, REG_XMM0, reg3);
```

Comment:
We should handle this in lsra and explicitly specify we need the XMM0 register.

Reply:
I tried! But I seem not able to set register candidates of a operand to XMM0 (only can do it for internal temp registers).  Could you help? cc @CarolEidt @mikedn

Reply:
If there would be several adjacent intrinsics, each requiring return register different from previously used return register than  `XMM0` will not be best candidate, unless I misunderstand how lsra will handle such operation.

Reply:
On machines which don't support the VEX encoding, the register has to be XMM0. It isn't optional.

![image](https://user-images.githubusercontent.com/10487869/34882892-c7c5960c-f76c-11e7-9843-d24ed6146415.png)

Reply:
I don't think we can merge before this is resolved. This is using a register that LSRA isn't aware we are using (we also aren't checking that `reg`, `reg1`, or `reg2` aren't XMM0, so we could be overwriting data used by the next couple lines).

Reply:
> I tried! But I seem not able to set register candidates of a operand to XMM0 (only can do it for internal temp registers)

What exactly have you tried? Doesn't `setDstCandidates(RBM_XMM0)` work?

Reply:
Right.

Reply:
Does "Right" means that you used `setDstCandidates`? Anyway, it's supposed to be `setSrcCandidates`, not sure why I wrote `Dst` instead.